### PR TITLE
dev/drupal#72 Ensure front end profile title is used in event confirmation screens & emails

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fixes a bug whereby the front end profile title is not used in event confirm & thankyou screens & confirmation emails

Before
----------------------------------------
Register ...
Screen Shot 2019-08-05 at 7 00 29 AM

BUT
Screen Shot 2019-08-05 at 7 00 51 AM

Also in confirmation emails

After
----------------------------------------
Screen Shot 2019-08-05 at 8 03 46 AM
Screen Shot 2019-08-05 at 8 04 10 AM

Technical Details
----------------------------------------
Incorporates https://github.com/civicrm/civicrm-core/pull/14958

Comments
----------------------------------------

https://lab.civicrm.org/dev/drupal/issues/72

### CODE REVIEW

**Comment by seamuslee001** [comment]
@jusfreeman you got any interest in this?

**Comment by jusfreeman** [comment]
@seamuslee001 yes, thanks

**Comment by eileenmcnaughton** [comment]
I should note I'm not actually personally vested in this - I was looking at test cover in order to build that up before fixing a more complex issue (which probably will require test build up & code cleanup before I could write a patch that I could reasonably expect someone else to review).

Fixing trivial bugs is usually a good way to add test cover & do relevant function extraction

File: CRM/Event/BAO/Event.php [hunk 1]
```diff
@@ -1446,11 +1454,9 @@ public static function displayProfile(&$params, $gid, &$groupTitle, &$values, &$
         $fields = CRM_Core_BAO_UFGroup::getFields($gid, FALSE, CRM_Core_Action::ADD);
       }
 
-      foreach ($fields as $v) {
-        if (!empty($v['groupTitle'])) {
-          $groupTitle['groupTitle'] = $v['groupTitle'];
-          break;
-        }
+      if (!empty($fields)) {
+        $profile = civicrm_api3('UFGroup', 'getsingle', ['id' => $gid, ['return' => ['title', 'frontend_title']]]);
```
Comment (seamuslee001) [inline CRM/Event/BAO/Event.php]: @eileenmcnaughton any reason not to use `CRM_UF_BAO_UFGroup::getFrontEndTitle()` here?

Comment (eileenmcnaughton) [reply CRM/Event/BAO/Event.php]: because it didn't exist when I wrote those lines....

**Comment by seamuslee001** [comment]
(*CiviCRM Review Template WORD-1.2*)



* General standards
    * ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain)) Pass
    * ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user)) Pass i tested this on an event registration and confirm that this fixes it, also confirmed that it changes it for the confirm step as well which makes it all in line
    * ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc)) Pass
    * ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run)) Pass as per #r-user 
* Developer standards
    * ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech)) Pass 
    * ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code)) Pass subject to answer on question
    * ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint)) Pass
    * ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test)) Pass

File: CRM/Event/BAO/Event.php (hunk 1 above)
Comment (seamuslee001) [reply CRM/Event/BAO/Event.php]: I think it would probably be good to tidy this up before merging but otherwise looks good to me

Comment (eileenmcnaughton) [reply CRM/Event/BAO/Event.php]: ok done

**Comment by seamuslee001** [comment]
Added MOP

**Comment by jusfreeman** [comment]
Happy to test this, but I need some guidance for how to test. Since I could not find any profile titles being output at all in the event registration confirmation emails.

**Comment by eileenmcnaughton** [comment]
in testing I was seeing the custom pre & post profile blocks had titles. Maybe test the rc since that gets cut tomorroe & this has mop

**Comment by eileenmcnaughton** [comment]
note this incorporates #14958 

also - my money says that contribution pages have this bug too...

**Comment by seamuslee001** [comment]
@eileenmcnaughton test fail looks related https://test.civicrm.org/job/CiviCRM-Core-PR/28196/testReport/junit/(root)/CRM_Event_Form_Registration_ConfirmTest/testAssignProfiles/

**Comment by eileenmcnaughton** [comment]
Ah yes...

**Pull request merged**